paddle/fluid/platform/mkldnn_utils.h

```diff
@@ -0,0 +1,183 @@
+/* Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+// NOTE:
+// GetMKLDNNFormat and ToMKLDNNDataType functions are here temporarily. They are
+// needed because without them forward declaration was causing an error when
+// building with "-DWITH_TESTING=ON". This file will be deleted after completing
+// md-related refactoring
+
+namespace paddle {
+namespace platform {
+
+inline dnnl::memory::format_tag GetMKLDNNFormat(dnnl::memory::desc mem_desc) {
```

Comment:
Maybe some functions to classify the order of the strides and values of the inner_ variables could be created, as they seem to follow a similar pattern? This would make it possible to transform this function into a map and make it much more readable.

Reply:
@piotrekobi the main goal of integrating memory descriptor into tensor class is to delete these ugly functions. After full refactoring they will be deleted since we won't be using format tags at all and we will rely on strides/offsets which are included in memory descriptor

Reply:
Makes sense. Thanks for the explanation.